src/main/java/net/greghaines/jesque/utils/JedisUtils.java

```diff
@@ -175,4 +188,30 @@ public static boolean canUseAsRecurringQueue(final Jedis jedis, final String que
     private JedisUtils() {
         // Utility class
     }
+
+    /**
+     * Determines if the queue identified by the given key can be used as a priority queue.
+     *
+     * @param jedis connection to Redis
+     * @param key   the key that identifies a queue  @return true if the key already is a delayed queue or is not currently used, false otherwise
+     */
+    public static boolean canUseAsPriorityQueue(final Jedis jedis, final String key) {
+        return useAsQueueOfType(jedis, key, PRIORITY_QUEUE);
+    }
+
+    private static boolean useAsQueueOfType(final Jedis jedis, final String key, final String queueType) {
+        final String existingType = jedis.hget(QUEUE_TYPES, key);
+        if (existingType == null) {
+            jedis.hset(QUEUE_TYPES, key, queueType);
```

Comment:
@itaim I believe there is a race condition here if two clients are attempting to create a queue of the same name but different type at the same time. Also, it looks like we never remove entries from this hash, which conflicts with the current semantics of reassigning the queue type as long as it is empty.